apps/explorer/src/pages/validator/ValidatorDetails.tsx

```diff
@@ -0,0 +1,208 @@
+
+import { DelegatorsList } from './DelegatorsList';
+
+import { ReactComponent as ArrowRight } from '~/assets/SVGIcons/12px/ArrowRight.svg';
+import ErrorResult from '~/components/error-result/ErrorResult';
+import { useGetObject } from '~/hooks/useGetObject';
+import {
+    VALIDATORS_OBJECT_ID,
+    type ValidatorState,
+} from '~/pages/validator/ValidatorDataTypes';
+import { Button } from '~/ui/Button';
+import { DescriptionList, DescriptionItem } from '~/ui/DescriptionList';
+import { Heading } from '~/ui/Heading';
+import { ImageIcon } from '~/ui/ImageIcon';
+import { AddressLink } from '~/ui/InternalLink';
+import { LoadingSpinner } from '~/ui/LoadingSpinner';
+import { TableHeader } from '~/ui/TableHeader';
+import { Text } from '~/ui/Text';
+import { getName } from '~/utils/getName';
+import { getStakedPercent } from '~/utils/getStakedPercent';
+
+export type Delegator = {
+    delegator: string;
+    sui_amount: bigint;
+    share: number;
+    type: string;
+};
+
+function ValidatorDetails() {
+    const { id } = useParams();
+
+    const { data, isLoading } = useGetObject(VALIDATORS_OBJECT_ID);
+
+    const validatorsData =
+        data && isSuiObject(data.details) && isSuiMoveObject(data.details.data)
+            ? (data.details.data.fields as ValidatorState)
+            : null;
+
+    const validatorData = useMemo(() => {
+        if (!validatorsData) return null;
+        return (
+            validatorsData.validators.fields.active_validators.find(
+                (av) => av.fields.metadata.fields.sui_address === id
+            ) || null
+        );
+    }, [id, validatorsData]);
+
+    const validator = useMemo(() => {
+        if (!validatorData || !validatorsData) return null;
+
+        const { name, pubkey_bytes, sui_address } =
+            validatorData.fields.metadata.fields;
+        const {
+            sui_balance,
+            starting_epoch,
+            pending_delegations,
+            delegation_token_supply,
+        } = validatorData.fields.delegation_staking_pool.fields;
+
+        const num_epochs_participated = validatorsData.epoch - starting_epoch;
+
+        const APY = Math.pow(
+            1 +
+                (sui_balance - delegation_token_supply.fields.value) /
+                    delegation_token_supply.fields.value,
+            365 / num_epochs_participated - 1
+        );
+
+        const totalStaked = [...pending_delegations]?.reduce(
+            (acc, fields) => (acc += BigInt(fields.fields.sui_amount || 0n)),
+            0n
+        );
+
+        return {
+            name: getName(name),
+            pubkeyBytes: new Base64DataBuffer(
+                new Uint8Array(pubkey_bytes)
+            ).toString(),
+            suiAddress: sui_address,
+            apy: APY > 0 ? APY : 'N/A',
+            logo: null,
+            link: null,
+            address: sui_address,
+            totalStaked,
+            delegators: Object.values(
+                [...pending_delegations]?.reduce(
```

Comment:
Before we merge this, we need to figure out what the plan is for delegators, given we can't display active delegators.